libs/api/domains/license-service/src/lib/client/machine-license-client/genericMachineLicense.module.ts

```diff
@@ -0,0 +1,34 @@
+import { AdrAndMachineLicenseClientModule } from '@island.is/clients/adr-and-machine-license'
+import {
+  SmartSolutionsApi,
+  SmartSolutionsApiClientModule,
+  SmartSolutionsConfig,
+  SMART_SOLUTIONS_API_CONFIG,
+} from '@island.is/clients/smartsolutions'
+import { Module } from '@nestjs/common'
+import { ConfigType } from '@nestjs/config'
+import { GenericFirearmLicenseConfig } from '../firearm-license-client'
+import { GenericMachineLicenseService } from './genericMachineLicense.service'
+import { GenericMachineLicenseConfig } from './genericMachineLicense.config'
+
+@Module({
+  imports: [SmartSolutionsApiClientModule, AdrAndMachineLicenseClientModule],
+  providers: [
+    GenericMachineLicenseService,
+    {
+      provide: SMART_SOLUTIONS_API_CONFIG,
+      useFactory: (config: ConfigType<typeof GenericMachineLicenseConfig>) => {
+        const smartConfig: SmartSolutionsConfig = {
+          apiKey: config.apiKey,
+          apiUrl: config.apiUrl,
+          passTemplateId: config.passTemplateId,
+        }
+        return smartConfig
+      },
+      inject: [GenericFirearmLicenseConfig.KEY],
+    },
+    SmartSolutionsApi,
+  ],
+  exports: [GenericMachineLicenseService],
+})
```

Comment:
Ah, I see that you're instantiating multiple instances of SmartSolutionApi, each with different configuration. I guess it is not the "easy" way of using our ConfigModule, but this is still better than before.

We should join this with the [proper way](https://docs.nestjs.com/fundamentals/dynamic-modules#community-guidelines) of "instantiating multiple instances" of modules.

Currently you're importing SmartSolutionsApiClientModule, but then re-providing its options and api. Instead we should do something like this:

```
  imports: [
    SmartSolutionsApiClientModule.registerAsync({
      useFactory: (config: ConfigType<typeof GenericMachineLicenseConfig>) => {
        const smartConfig: SmartSolutionsConfig = {
          apiKey: config.apiKey,
          apiUrl: config.apiUrl,
          passTemplateId: config.passTemplateId,
        }
        return smartConfig
      },
      inject: [GenericFirearmLicenseConfig.KEY],
    }),
  ]
```

It would be nice to have NestJS 9's new `ConfigurableModuleBuilder` utility to get `registerAsync` for free. Until then, you basically need something like this:

```
@Module(...)
class SmartSolutionsApiClientModule {
  register(config: SmartSolutionsConfig) {
    return {
      module: SmartSolutionsApiClientModule,
      providers: [{ provide: SMART_SOLUTIONS_API_CONFIG, value: config }],
    }
  }

  registerAsync(configProvider: ?) {
    return {
      module: SmartSolutionsApiClientModule,
      providers: [{ ...configProvider, provide: SMART_SOLUTIONS_API_CONFIG }],
    }
  }
}
```

Reply:
Ah ok, will do